docs/source/working_with_aiida/caching.rst

```diff
@@ -4,45 +4,90 @@
+There are numerous reasons why you may need to re-run calculations you’ve already done before.
+Since AiiDA stores the full provenance of each calculation, it can *detect* whether a calculation has been run before
+and reuse its outputs without wasting computational resources.
 
-Caching is also implemented for Data nodes. This is not very useful in practice (yet), but is an easy way to show how the caching mechanism works:
+This feature is not (yet) enabled by default. In order to enable hashing:
 
-.. ipython::
-    :verbatim:
+ 1. Place the following ``cache_config.yml`` file in your ``.aiida`` configuration folder
+    in order to enable caching for your AiiDA profile (here called ``aiida2``):
 
-    In [1]: from __future__ import print_function
+    .. code:: yaml
 
-    In [2]: from aiida.orm import Str
+        aiida2:
+          default: True
 
-    In [3]: n1 = Str('test string')
+ 2. Run ``verdi -p aiida2 rehash`` in order to compute hashes of all nodes in your existing database
 
-    In [4]: n1.store()
-    Out[4]: u'test string'
+From this point onwards, when you submit a new calculation, AiiDA will compute hashes of the type of calculation and its inputs (see :ref:`caching_matches`). If the hash of the calculation is found to be already present in the database, AiiDA will reuse its results without repeating the actual calculation.
 
-    In [5]: n2 = Str('test string')
+.. _fig_caching:
+.. figure:: include/images/caching.png
+  :align: center
+  :height: 350px
 
-    In [6]: n2.store(use_cache=True)
-    Out[6]: u'test string'
+  When resusing a cached calculation, AiiDA will link up to the input nodes as usual, and make a copy of both the calculation node **C** and its outputs **D3**.
 
-    In [7]: print('UUID of n1:', n1.uuid)
-    UUID of n1: 956109e1-4382-4240-a711-2a4f3b522122
+In order to ensure that the provenance graph with and without caching is the same,
+AiiDA creates a *copy* of the orginial calculation node, linked to the original inputs, as well as a to a copy of the original outputs, as shown in :numref:`fig_caching`.
 
-    In [8]: print('n2 is cached from:', n2.get_extra('_aiida_cached_from'))
-    n2 is cached from: 956109e1-4382-4240-a711-2a4f3b522122
+.. note:: Caching is **not** implemented at the WorkChain/workfunction level (see :ref:`caching_limitations` for details).
+
+
+.. _caching_matches:
 
-As you can see, passing ``use_cache=True`` to the ``store`` method enables using the cache. The fact that ``n2`` was created from ``n1`` is stored in the ``_aiida_cached_from`` extra of ``n2``.
+How are nodes hashed?
+---------------------
+
+By default, the hash of a node is computed from:
+
+* all attributes of the node, except the ``_updatable_attributes``
```

Comment:
This also includes the `_hash_ignored_attributes` (currently only defined for `CalcJob`)

Reply:
I've added this now.